src/pybind/mgr/rbd_support/module.py

```diff
@@ -1185,10 +1210,12 @@ def queue_migration_abort(self, image_spec):
     def task_cancel(self, task_id):
         self.log.info("task_cancel: {}".format(task_id))
 
-        if task_id not in self.tasks_by_id:
+        task = self.tasks_by_id.get(task_id)
+        if not task or not is_authorized(self.module,
+                                         task.refs[TASK_REF_POOL_NAME],
+                                         task.refs[TASK_REF_POOL_NAMESPACE]):
```

Comment:
nit: shouldn't we rather return `EACCES` when it is not authorized? If such a situation really happened it would be misleading to get `ENOENT`.

Reply:
I had that in the first implementation, but then decided to avoid leaking information. Since the list strips out unauthorized tasks from the list, to me it seemed like this mirrored that idea.

Reply:
The list is a little different case to me -- when you don't see the tasks in the list because you are not allowed. But when someone gets a task by some way and wants to cancel it and the mgr tells that this task does not exist, she would think that the task already finished or did not existed at all, while the task would still be running.

Anyway, it is not very likely scenario and I don't expect many problems with this, so I am fine with any variant.